src/Azure.Functions.Cli/Actions/HostActions/StartHostAction.cs

```diff
@@ -652,6 +659,31 @@ private static string GetInProcNet6ExecutablePath()
             return Path.Combine(funcExecutableDirectory, InProc6DirectoryName, executableName);
         }
 
+        private bool IsInvalidInProc6SdkVersion()
+        {
+            var functionAppRoot = ScriptHostHelpers.GetFunctionAppRootDirectory(Environment.CurrentDirectory);
+            string projectFilePath = ProjectHelpers.FindProjectFile(functionAppRoot);
```

Comment:
Should we be relying on the project file like this? Will this resolve properly in VS? I wonder if we should instead be using the built assets.

Reply:
Yes, the goal here should be to evaluate the application state (inspect whether you're running an in-proc payload against the OOP Core Tools). Relying on the parsing the project to perform an SDK check here will be unreliable (e.g., using Central Package Management, importing packages from other projects/files, etc.)

@surgupta-msft we should be able to inspect the runtime configuration and detect if it's an in-proc payload.

Reply:
Do we only want to detect if OOP coretools is getting inproc6 payload? or do we also want to detect if that inproc6 payload is referencing to the older version of `Microsoft.NET.Sdk.Functions` which does not support custom host v0?

If it is option1 then I don't need to search for sdk and its version in the payload. But if it is option2 and we dont want to rely on project file and then may be I can parse `deps.json` file to find out the sdk version that will be loaded. Let me know if there is a better way to know the specific package version

Reply:
Ah, my apologies, Fabio's comment here got deleted by mistake :( Not able to undo it. Had an offline discussion with Fabio. Our goal is to detect if OOP host is getting inproc payload and log the message to update the sdk to the latest and exit.